flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -532,6 +532,51 @@ public class ExecutionConfigOptions {
                                     + "the correctness of the results.\n"
                                     + "ORDERED ensures that the operator emits the result in the same order as the data enters it. This is the default.");
 
+    // ------------------------------------------------------------------------
+    //  Async VECTOR_SEARCH Options
+    // ------------------------------------------------------------------------
+    @Documentation.TableOption(execMode = Documentation.ExecMode.BATCH_STREAMING)
+    public static final ConfigOption<Boolean> TABLE_EXEC_ASYNC_VECTOR_SEARCH_ASYNC =
+            key("table.exec.async-vector-search.async")
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "Whether to run an async search function or not. Default to false.");
```

Comment:
I think this is ambiguous. Does it mean:
`when false the function does not run` or `when false the function runs synchronously.`